tests/scripts/thread-cert/thread_cert.py

```diff
@@ -527,35 +553,46 @@ def _has_backbone_traffic(self):
 
     def _prepare_backbone_network(self):
         """
-        Prepares one or multiple backbone network(s).
+        Create one or multiple backbone networks(docker bridge networks) based on the TOPOLOGY definition.
+
+        1. If there's `backbone_network_id` defined in the TOPOLOGY, it creates the backbone network based on the value.
+           The format of the
+             * backbone name   is `backbone{PORT_OFFSET}.{backbone_network_id}`,      for example, "backbone0.0", "backbone0.1";
+             * backbone prefix is `backbone{PORT_OFFSET}:{backbone_network_id}::/64`, for example, "9100:0::/64", "9100:1::/64".
 
-        This method creates the backbone network based on the `backbone` values defined in the TOPOLOGY in each test.
-        If no backbone values are defined, it sets the default backbone network as BACKBONE_DOCKER_NETWORK_NAME
-        from the config module.
+        2. If no `backbone_network_id` is defined, it creates the default backbone network, the format of the
+             * backbone name   is `backbone{PORT_OFFSET}.0`,    for example, "backbone0.0";
+             * backbone prefix is `backbone{PORT_OFFSET}::/64`, for example, "9100::/64".
         """
-        # Use backbone_set to store all the backbone values in TOPOLOGY
+        # Create backbone_set to store all the (backbone_name, backbone_prefix) pairs by parsing TOPOLOGY.
         backbone_set = set()
         for node in self.TOPOLOGY:
-            backbone = self.TOPOLOGY[node].get('backbone_network')
-            if backbone:
-                backbone_set.add(backbone)
+            backbone_id = self.TOPOLOGY[node].get('backbone_network_id')
+            if backbone_id is not None:
+                backbone_set.add((f'{config.BACKBONE_DOCKER_NETWORK_NAME}.{backbone_id}',
+                                  f'{config.BACKBONE_IPV6_ADDR_START}:{backbone_id}::/64'))
 
-        # Set default backbone network if backbone_set is empty
+        # Set default backbone network name and prefix if backbone_set is empty
         if not backbone_set:
-            backbone_set.add(config.BACKBONE_DOCKER_NETWORK_NAME)
+            backbone_set.add((f'{config.BACKBONE_DOCKER_NETWORK_NAME}.{config.BACKBONE_DOCKER_NETWORK_DEFAULT_ID}',
+                              f'{config.BACKBONE_IPV6_ADDR_START}::/64'))
 
         # Iterate over the backbone_set and create backbone network(s)
-        for offset, backbone in enumerate(backbone_set, start=PORT_OFFSET):
-            backbone_prefix = f'{config.BACKBONE_IPV6_ADDR_START_BASE + offset:04x}::/64'
+        for backbone, backbone_prefix in backbone_set:
+            self._backbone_network_names.append(backbone)
             self.assure_run_ok(
                 f'docker network create --driver bridge --ipv6 --subnet {backbone_prefix} -o "com.docker.network.bridge.name"="{backbone}" {backbone} || true',
                 shell=True)
```

Comment:
A simpler way to achieve this is to create a set of backbone network IDs and do the string manipulation inside the loop.

Then you don't have to repeat the string manipulations in code.

Reply:
good suggestion! I will update

Reply:
updated, thanks!